src/dataBase/postgres/postgresBookRepo.go

```diff
@@ -38,16 +44,18 @@ func GetByID(bookID int) (repository.Book, error) {
 	if err := rows.Err(); err != nil {
 		log.Fatal(err)
 	}
-	return book, nil
+	return
 
 }
 
-func GetAll() ([]interface{}, error){
+
+//GetAll iterates over the DB using the SQL SELECT Request and return all books from DB
+func (p *booksRepositoryPG) GetAll() ([]interface{}, error) {
```

Comment:
If `booksRepositoryPG` implements some interface, the receiver shouldn't be a pointer. See https://play.golang.org/p/x0tRkfDe4Yu

Reply:
Thanks a lot!